webapp/views.py

```diff
@@ -95,3 +106,182 @@ def get(self, request, *args, **kwargs):
             **kwargs
         )
         return self.render_to_response(context)
+
+
+class SearchView(TemplateView):
+    '''
+    Return search results from the Google Search Appliance
+
+    Requests should be formatted: <url>?q=<string>&offset=<num>&limit=<num>
+
+    I've used "offset" and "limit" for pagination,
+    following the "Web API Design" standard:
+    https://pages.apigee.com/web-api-design-ebook.html
+
+    This gets results from Canonical's Google Search Appliance,
+    currently located at: butlerov.internal (10.22.112.8)
+    '''
+
+    template_name = "pages/search.html"
+
+    def get_context_data(self, **kwargs):
+        """
+        Extend CMSPageView.get_context_data to parse query parameters
+        and return search results from the Google Search Appliance (GSA)
+
+        E.g.: http://example.com/search?q=juju&limit=10&offset=10
+
+        Query parameters:
+        - q: the search query to be passed to the GSA
+        - limit: number of results to return, "page size" (default: 10)
+        - offset: where to start results at (default: 0)
+        """
+
+        # On live the GSA domain will be butlerov.internal
+        # but on dev, we need to access GSA through localhost
+        # (see GSASearchView docstring above)
+        gsa_domain = 'butlerov.internal'
+
+        parser = GSAParser(gsa_domain)
+
+        # Import context from parent
+        context = super(SearchView, self).get_context_data(**kwargs)
+
+        # defaults + GET params
+        context.update({
+            'query': self.request.GET.get('q', ''),
+            'results': [],
+            'request_succeeded': True,
+            'parse_succeeded': True,
+            'start': 0,
+            'end': 0,
+            'limit': int(self.request.GET.get('limit', '10')),
+            'offset': int(self.request.GET.get('offset', '0')),
+            'total': 0,
+            'nav_items': []
+        })
+
+        # return self.context
+        try:
+            if getattr(settings, 'DEBUG', False):
```

Comment:
I don't think you need to specify the default argument here. It defaults to None which works for the if statement.